### ISSUE TITLE
Open Match does not refresh Match Function Client Cache

### ISSUE THREAD

**Issue by sawagh** [googleforgames/open-match#917]
Open Match Backend Service caches match function clients. If one of the client is stale, it does not refresh the cache upon inability to connect. I hit this in the following scenario:

1. I deployed OM Core, and a Match Function - used the director to run E2E successfully,
2. I updated my Match Function Image, pushed and refreshed my Match Function Service
3. This caused Match Function Service to be directed to a different IP
4. Open Match failed to connect to the new service as it was attempting to connect to the cached one.

**Comment by Laremere** [comment]
This seems to be a problem with dns caching, or with a client connection which is not properly terminated when the server it is connected to stops.

**Comment by yfei1** [comment]
@sawagh I couldn't reproduce this bug. Here are the steps that I used:

1. use `make install-chart` to install open match with the demo
2. make sure everything work in `make proxy`
3. make some changes to the `om-function` source code
4. rebuild the image with a new tag
5. update the deployment `spec.template.spec.containers[0].image` to the new image tag
6. the demo still works and I was able to see the changes online from `kubectl log -n open-match-demo om-function`
